cmd/kubectl-directpv/migrate.go

```diff
@@ -0,0 +1,382 @@
+			// https://stackoverflow.com/questions/9251234/how-to-check-the-uniqueness-inside-a-for-loop
+			if _, ok := setOfUniqueFSUUIDs[NewDrive.Status.FSUUID]; ok {
+				fmt.Println("FSUUID found already, we can't convert this drive unless is unique: ", NewDrive.Status.FSUUID)
+			} else {
+				fmt.Println("element not found, FSUUID is unique, proceed to be added/appended for conversion")
+				setOfUniqueFSUUIDs[NewDrive.Status.FSUUID] = struct{}{}
+				// Append the new drive to the list
+				newDrives = append(newDrives, NewDrive)
+				driveFsuuid[NewDrive.ObjectMeta.Name] = NewDrive.Status.FSUUID
+			}
+		}
+	}
+
+	// Return the new list of drives
+	return newDrives, nil
+}
+
+func migrateMain(ctx context.Context) {
+	k8s.Init()
+
+	var err error
+	if directCSIDriveClient, err = directcsiclient.DirectCSIDriveInterfaceForConfig(k8s.KubeConfig()); err != nil {
+		klog.Fatalf("unable to create new direct-csi drive interface; %v", err)
+	}
+
+	if directCSIVolumeClient, err = directcsiclient.DirectCSIVolumeInterfaceForConfig(k8s.KubeConfig()); err != nil {
+		klog.Fatalf("unable to create new direct-csi volume interface; %v", err)
+	}
+
+	gvk, err := directcsiclient.GetGroupKindVersions(
+		DirectCSIGroup,
+		"DirectCSIDrive",
+		"v1beta5",
+		directv1beta4.Version,
+		directv1beta3.Version,
+		directv1beta2.Version,
+		directv1beta1.Version,
+		directv1alpha1.Version,
+	)
+	if err != nil {
+		fmt.Println("migrateMain: unable to get GetGroupKindVersions of direct-csi: ", err)
+		return
+	}
+
+	if gvk.Group != DirectCSIGroup {
+		klog.Fatalf("migration does not support direct-csi group %v", gvk.Group)
+	}
+
+	switch gvk.Version {
+	case "v1beta3", "v1beta4", "v1beta5":
+	default:
+		klog.Fatalf("migration does not support direct-csi version %v", gvk.Version)
+	}
+
+	// Then we list the drives
+	// This is getting all the info from all the drives
+	ctx, cancelCtx := context.WithCancel(context.Background())
+	defer cancelCtx()
+
+	filteredDrives := []directcsi.DirectCSIDrive{}
+	for result := range ListDrives(ctx) {
+		if result.Err != nil {
+			klog.Fatalf("unable to get direct-csi drives; %v", result.Err)
+		}
+
+		// FIXME: ignore result.Drive if its source version is < v1beta3
+
+		if !uuidRegex.MatchString(result.Drive.Status.FilesystemUUID) {
+			klog.Fatalf("invalid FilesystemUUID %v in DirectCSIDrive %v", result.Drive.Status.FilesystemUUID, result.Drive.Name)
+		}
+
+		for _, finalizer := range result.Drive.Finalizers {
+			if !strings.Contains(finalizer, DirectCSIGroup) {
+				klog.Fatalf("invalid finalizer value %v in DirectCSIDrive %v", finalizer, result.Drive.Name)
+			}
+		}
+
+		filteredDrives = append(filteredDrives, result.Drive)
+	}
+
+	// To convert from old CRDs to new CRDs
+	newDriveCRD, _ := migrateDriveCRD(filteredDrives)
+	// Output Drives as YAML
+	for index, drive := range newDriveCRD {
+		// Output as YAML
+		yamlName := "drive-" + strconv.Itoa(index+1) + ".yaml"
```

Comment:
we don't need to create one yaml file for each drive. We can just create one file (drives.yaml) and write all the drive manifests there. Seperate by `\n---\n`.